lib/net/winrm/ctrl_c.rb

```diff
@@ -1,47 +1,84 @@
-# WSMV message to send a Ctrl+C signal to a remote shell
 require 'winrm/wsmv/base'
+
 module Net
   module MsfWinRM
-    # A WinRM Ctrl+C signal message
+    # Represents a WinRM Ctrl+C signal message
     class CtrlC < WinRM::WSMV::Base
+      # Initializes the CtrlC message
+      #
+      # @param session_opts [Hash] Options for WinRM session
+      # @param opts [Hash] Additional options including :shell_id and :command_id
+      # @option opts [String] :shell_id The WinRM shell identifier
+      # @option opts [String] :command_id The WinRM command identifier
+      # @option opts [String] :shell_uri Optional URI for shell
+      #
+      # @raise [ArgumentError] If :shell_id or :command_id is missing
       def initialize(session_opts, opts)
-        raise 'opts[:shell_id] is required' unless opts[:shell_id]
-        raise 'opts[:command_id] is required' unless opts[:command_id]
-
+        validate_options(opts)
         super()
-
         @session_opts = session_opts
         @shell_id = opts[:shell_id]
         @command_id = opts[:command_id]
         @shell_uri = opts[:shell_uri] || RESOURCE_URI_CMD
       end
 
-      protected
+      # Sends the Ctrl+C signal
+      #
+      # @param async [Boolean] Whether to send signal asynchronously (default: false)
+      # @param timeout [Integer] Timeout in seconds for the operation (default: nil)
+      #
+      # @return [Boolean] True if signal sent successfully, false otherwise
+      def send_signal(async: false, timeout: nil)
```

Comment:
I cannot find where `send_signal` is called from. I might be missing something. Please, can you elaborate?